### DISCUSSION TITLE
Lack of contact despite enforced glued contact

### DISCUSSION THREAD

**Discussion by klaw13** [idaholab/moose#17591] [Q&A Modules: General] [answered]
Hello,

I've been running into some problems with getting the following volumes to stay in contact with one another. 
![image](https://user-images.githubusercontent.com/20307070/114490184-042a7800-9bda-11eb-895c-772d0016f749.png)

I'm using the contact module and the glued model, but the volumes detached from one another during simulation when the four cylindrical components are pressurized. 

I then constrained the variables (disp_x, disp_y, and disp_z) on each side of each volume using the EqualValueBoundaryConstraint. In Paraview, it seems that the parts now stay together, but the simulation still generates the following message:  

```
*** Warning ***
Warning in PenetrationLocator. Penetration is not detected for one or more secondary nodes. This could be because those secondary nodes simply do not project to faces on the primary surface. However, this could also be because contact should be enforced on those nodes, but the faces that they project to are outside the contact patch, which will give an erroneous result. Use appropriate options for 'patch_size' and 'patch_update_strategy' in the Mesh block to avoid this issue. Setting 'patch_update_strategy=iteration' is recommended because it completely avoids this potential issue. Also note that this warning is printed only once, so a similar situation could occur multiple times during the simulation but this warning is printed only at the first occurrence." 
```

Does anyone have any suggestions as to how to resolve this? 

Thanks! 

Kelsey

**Comment by GiudGiud** [comment, marked as answer]
@lindsayad @bwspenc on constraints / contact

**Reply by bwspenc** [reply]
My first thought is that the blocks might not actually be touching each other initially.

**Reply by makeclean** [reply]
Could you share your inputs?

**Reply by klaw13** [reply]
Thanks for all your input! Seems our system was weakly constrained. We put in additional boundary conditions at the points of contact and were able to resolve the issue.

**Reply by jbadger95** [reply]
@klaw13 3D mortar constraints are coming soon, still putting the finishing touches on it and the contact kernels will need some tweaks but in case you're interested [#18203](https://github.com/idaholab/moose/pull/18203).

**Reply by klaw13** [reply]
Good to know, thanks @jbadger95!
